scripts/cpplint.py

```diff
@@ -3974,13 +3974,6 @@ def CheckBracesSpacing(filename, clean_lines, linenum, nesting_state, error):
     for offset in xrange(endlinenum + 1,
                          min(endlinenum + 3, clean_lines.NumLines() - 1)):
       trailing_text += clean_lines.elided[offset]
-    # We also suppress warnings for `uint64_t{expression}` etc., as the style
-    # guide recommends brace initialization for integral types to avoid
-    # overflow/truncation.
-    if (not Match(r'^[\s}]*[{.;,)<>\]:]', trailing_text)
-        and not _IsType(clean_lines, nesting_state, leading_text)):
-      error(filename, linenum, 'whitespace/braces', 5,
-            'Missing space before {')
```

Comment:
Maybe removing this is the right thing to do, but then we should remove all the code computing `leading_text`, `trailing_text` above. Or maybe you should just use `NOLINTNEXTLINE(whitespace/braces)` in the test.

Reply:
`clang-format` already checks for positions and spacing of curly braces and parentheses. The problem stems from the fact this isn't aware of initializer lists/uniform initialization.
```c++
std::vector<std::pair<int, int>>{{1, 2}, {3, 4}, {5, 6}};
```
In the days of C++98, the above wasn't a valid syntax. Curly braces would appear only after `if` /`for` statements, as block scopes or surround function bodies. The linter expects this to still be true.

Reply:
Ok, in which case the linter should be cleaned properly so as not to leave in place unused code.